### ISSUE TITLE
Not working on relibc

### ISSUE THREAD

**Issue by bjorn3** [rust-analyzer/rust-analyzer#5268]
https://gitlab.redox-os.org/redox-os/relibc/

Relibc has an unusual structure: Inside the `src` dir there are multiple dirs which each contain a regular cargo project with `Cargo.toml` and `src/lib.rs`, but also a `./mod.rs`, which is used from the top-level `src/lib.rs`. For these `mod.rs` ide features are not working.

**Comment by matklad** [comment]
Yeah, this does not work because we currently softly assume that modules can't point across crates (we used this to avoid checking each `mod foo;` in stdlib, when a user crates new file in src). The solution here is to put all of the crates into a single FileSet, (instead of creating a fileset per crate). At the moment, we infer file sets from Cargo.toml though, and don't have a way to override them.
